pkg/ebpf/compiler/compiler.cpp

```diff
@@ -0,0 +1,273 @@
+#include "compiler.h"
+#include "files.h"
+
+#include <iostream>
+#include <mutex>
+#include <llvm/IR/LegacyPassManager.h>
+#include <llvm/Support/TargetSelect.h>
+#include <clang/CodeGen/CodeGenAction.h>
+#include <clang/Driver/Job.h>
+#include <clang/Driver/Tool.h>
+#include <clang/Frontend/CompilerInstance.h>
+#include <clang/Lex/PreprocessorOptions.h>
+
+std::once_flag ClangCompiler::llvmInitialized;
+
+enum Architecture { PPC, PPCLE, S390X, ARM64, X86 };
+
+namespace {
+    std::map<std::string, std::unique_ptr<llvm::MemoryBuffer>> remapped_files_;
+}
+
+ClangCompiler::ClangCompiler(const char *name) :
+    llvmContext(std::make_unique<llvm::LLVMContext>()),
+    diagOpts(new clang::DiagnosticOptions()),
+    errStream(errString),
+    textDiagnosticPrinter(new clang::TextDiagnosticPrinter(errStream, diagOpts.get())),
+    diagnosticsEngine(new clang::DiagnosticsEngine(diagID, diagOpts, textDiagnosticPrinter.get(), false)),
+    defaultCflags({
+        "clang", // DO NOT REMOVE, first flag is ignored
+        "-emit-llvm",
+        "-O2",
+        "-D__KERNEL__",
+        "-fno-color-diagnostics",
+        "-fno-unwind-tables",
+        "-fno-asynchronous-unwind-tables",
+        "-fno-stack-protector",
+        "-nostdinc",
+        "-isystem/virtual/lib/clang/include",
```

Comment:
is this a 'real' path ?

Reply:
No, hence the `virtual` at the beginning. This is how the embedded files (like `stdarg.h`) get used.